### ISSUE TITLE
Eliminate use of javax.servlet-api

### ISSUE THREAD

**Issue by patrickjamesbarry** [mock-server/mockserver#1860]
The version of spring this lib was currently pinning the java version to is now EOL.  The new version of spring requires jdk 17 and moves to jakarta instead of javax.  We need a new version of this lib that supports spring 6 and jakarta packages.  This would eliminate this lib from pullling in things like javax.servlet-api

**Comment by MBoegers** [comment]
Do you mean the Spring version, which needs bumb? If so, I could help to utilize OpenRewrite for this upgrade.

**Comment by wakingrufus** [comment]
I think the issue is all of the usages of javax.servlet (https://github.com/search?q=repo%3Amock-server%2Fmockserver+%22javax.servlet%22&type=code) would need to be migrated to jakarta.servlet

**Comment by MBoegers** [comment]
Indeed, nearly every usage of javax has to be migrated. But OpenRewrite also has tools for javax.servlet.

**Comment by jamesdbloom** [comment]
## Closing as Won't Fix

MockServer maintains **Java 11 as the minimum supported version** to maximize compatibility. According to our compatibility policy, approximately 23% of Java projects still run on Java 11, making it important to maintain broad compatibility.

### Why This Request Cannot Be Accepted

Migrating from `javax.servlet` to `jakarta.servlet` would require:

1. **Java 17+ minimum** — Spring 6 and Jakarta EE 9+ require Java 17 as the minimum version
2. **Breaking change** — The `javax` → `jakarta` namespace migration is a breaking change that would force all users to upgrade
3. **Ecosystem impact** — Many enterprise environments and long-term support projects cannot easily upgrade to Java 17+

### Current Java Compatibility Policy

From `AGENTS.md`:

> MockServer targets **Java 11** as the minimum supported version. This is a deliberate decision to maximise compatibility — approximately 23% of Java projects still run on Java 11.
>
> **Rules:**
> - The Maven compiler source/target MUST remain at `11`
> - NEVER accept dependency upgrades that require Java 17+
> - NEVER use Java language features or APIs introduced after Java 11
> - The `javax` namespace is used throughout — do NOT migrate to `jakarta` namespace

### Current Versions

- **Java**: 11+ (source/target in `pom.xml`)
- **Spring**: 5.3.39 (highest Spring 5.x compatible with Java 11 + `javax`)
- **Servlet API**: `javax.servlet` namespace

### Future Consideration

This change could be considered for a **future major version (e.g., MockServer 6.0)** that explicitly breaks backward compatibility and requires Java 17+. However, there are no current plans for such a release.

If you need to use Spring 6 or Jakarta EE 9+ in your project, you can:

1. Use dependency exclusions to exclude MockServer's transitive `javax.servlet-api` dependency
2. Run MockServer in a separate process/container rather than embedding it in your application
3. Use the standalone MockServer Docker image or JAR

Closing this issue as **won't fix** due to Java 11 compatibility requirements.

**Comment by durcon** [comment]
You are right, some Java 11 versions (free or only with commercial support) are still there:

- Orcale's (free) implementation is already EOL, see https://endoflife.date/oracle-jdk
- Eclipse Temurin will be EOL 31.10.2027, see https://endoflife.date/eclipse-temurin
- Amazon Corretto will be EOL 31.01.2032, see https://endoflife.date/amazon-corretto

but Spring's and Spring Boot's free version are already EOL:

- https://endoflife.date/spring-framework
- https://endoflife.date/spring-boot
